mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -500,8 +531,13 @@ class RegionBuilderHelper {
         return builder.create<complex::SubOp>(arg0.getLoc(), arg0, arg1);
       if (allFloatingPoint)
         return builder.create<arith::SubFOp>(arg0.getLoc(), arg0, arg1);
-      if (allBool)
+      if (allBool) {
+        if (emitError) {
```

Comment:
There's a lot of repetition of this pattern. Could we perhaps simplify into an extern function?

Reply:
There is a return inside a if statement, how would you simplify this? I could see it with a macro maybe but I'm not sure how to do this with a function.
(the error message is also slightly different between all the cases below)

Reply:
What about something like:
```c++
// Mostly helpful to the constant message ones
Value returnError(StringRef& msg) {
  emitError() << msg;
  return nullptr;
}
```

then use:
```c++
if (emitError)
  return returnError("message here");
llvm_unreachable("message here");
```

Reply:
I just looked into this, but this wouldn't be universally applicable in this file: some of the cases are streaming multiple piece together to form the error message, and some function that have this pattern are not returning a Value.